src/shared/libfido2-util.c

```diff
@@ -362,6 +365,26 @@ static int fido2_is_cred_in_specific_token(
         }
 }
 
+static void plymouth_start_interaction(const char *text, bool *ret_displayed) {
+        assert(ret_displayed);
+
+        if (plymouth_send_msg(text) < 0)
+                return;
+
+        *ret_displayed = true;
+}
+
+static void plymouth_end_interaction(bool *displayed) {
+        assert(displayed);
+
+        if (!*displayed)
+                return;
+
+        /* In theory 'm' should hide a message, but it doesn't work (long standing issue).
+         * As a workaround, sending a single NUL byte hides the previous messages. */
+        plymouth_send_msg("");
```

Comment:
In the previous version, you used `a` rather than `A`, but now this uses `A`. Is it intentional?

Or, maybe better to introduce a boolean flag to plymouth_send_msg() to control 'a' or 'A' ?